src/frontend/web_application/src/components/Dropdown/index.jsx

```diff
@@ -86,26 +87,31 @@ class Dropdown extends Component {
     this.dropdownControl = document.getElementById(`${CONTROL_PREFIX}-${this.props.id}`);
     this.toggle(this.props.show);
 
-    this.unsubscribeClickEvent = addEventListener('click', (ev) => {
-      const target = ev.target;
-      const exceptRefs = this.props.closeOnClickExceptRefs;
+    if (this.props.closeOnClick) {
+      this.unsubscribeClickEvent = addEventListener('click', (ev) => {
+        const target = ev.target;
+        // const exceptRefs = this.props.closeOnClickExceptRefs;
 
-      const dropdownClick = !this.props.closeOnClick &&
-        (this.dropdown === target || this.dropdown.contains(target));
-      const exeptRefsClick = exceptRefs && exceptRefs.find(ref => (ref === target));
-      const controlClick = this.dropdownControl &&
-        (this.dropdownControl === target || this.dropdownControl.contains(target));
+        const dropdownClick = (this.props.closeOnClick === 'exceptSelf') &&
```

Comment:
I prefer when the variable describe correctly what happens: if the click is on the dropdown but closeOnClick is something else, `dropdownClick` is set to false that is incorrect.

I think it's better to add this later in the condition (or rename the variable to something very weird):

```
if (this.props.closeOnClick === 'exceptSelf' && dropdownClick) { return; }
```